client/client-multi/client-c/build.sh

```diff
@@ -52,88 +52,28 @@ build() {
     cd ..
 }
 
-execute_tests() {
-    cd build-posix
-    ctest --output-on-failure .
-    cd ..
-}
-
-check_installed_software() {
-    if hash rats 2>/dev/null
-    then
-        RATS_INSTALLED=1
-    else
-        echo "Rats is not installed, skipping..."
-        RATS_INSTALLED=0
-    fi
-
-    if hash cppcheck 2>/dev/null
-    then
-        CPPCHECK_INSTALLED=1
-    else
-        CPPCHECK_INSTALLED=0
-        echo "cppcheck not installed, skipping..."
-    fi
-
-
-    if hash valgrind 2>/dev/null
-    then
-        VALGRIND_INSTALLED=1
-    else
-        VALGRIND_INSTALLED=0
-        echo "valgrind not installed, skipping..."
-    fi
-}
-
-run_valgrind() {
-    echo "Starting valgrind..."
-    cd build-posix
-    if [ ! -d valgrindReports ]
-    then
-        mkdir valgrindReports
-    fi
-
-    # CMake supports running memory checker (like valgrind) only as a step
-    # of CDash.
-    # Calling valgrind externally in relation to CTest is viable workaround.
-    # Possibly, this will be moved someday into the Kaa build system in a form
-    # of a cmake script.
-    valgrind --leak-check=full --show-reachable=yes --trace-children=yes -v \
-    --log-file=valgrind.log --xml=yes --xml-file=valgrindReports/%p.memreport.xml \
-    ctest --output-on-failure
-
-    cd ..
-    echo "Valgrind analysis finished."
-}
-
-run_cppcheck() {
-    echo "Starting Cppcheck..."
-    cppcheck --enable=all --std=c99 --xml --suppress=unusedFunction src/ test/ 2>build-posix/cppcheck_.xml > build-posix/cppcheck.log
-    sed 's@file=\"@file=\"client\/client-multi\/client-c\/@g' build-posix/cppcheck_.xml > build-posix/cppcheck.xml
-    rm build-posix/cppcheck_.xml
-    echo "Cppcheck analysis finished."
-}
-
 run_rats() {
     echo "Starting RATS..."
     rats --xml `find src/ -name *.[ch]` > build-posix/rats-report.xml
     echo "RATS analysis finished."
 }
 
 run_analysis() {
-    check_installed_software
-
-    if [ $VALGRIND_INSTALLED -eq 1 ]; then
-        run_valgrind
-    fi
-
-    if [ $CPPCHECK_INSTALLED -eq 1 ]; then
-        run_cppcheck
-    fi
-
-    if [ $RATS_INSTALLED -eq 1 ]; then
+    cd build-posix
+    ctest -T memcheck
```

Comment:
It is a bad practice. CMake suggests for these purposes ctest which has build-in functionality, so we needn't to add more pseudo targets to project.